src/libtiled/map.cpp

```diff
@@ -298,6 +298,18 @@ bool Map::addTemplateGroup(TemplateGroup *templateGroup)
     return true;
 }
 
+void Map::addTemplateGroupReference(const TemplateGroup *templateGroup)
+{
+//    mTemplateGroupReferences[templateGroup]++;
+    mTemplateGroupReferences.insert(templateGroup, mTemplateGroupReferences.value(templateGroup) + 1);
+}
+
+void Map::removeTemplateGroupReference(const TemplateGroup *templateGroup)
+{
+//    mTemplateGroupReferences[templateGroup]--;
```

Comment:
It worked fine but I wasn't sure which style to use. I will try the suggested approach it should be simpler.